source/Engine.cpp

```diff
@@ -2045,15 +2043,20 @@ void Engine::HandleMouseClicks()
 void Engine::HandleMouseInput(Command &activeCommands)
 {
 	isMouseHoldEnabled = activeCommands.Has(Command::MOUSE_TURNING_HOLD);
+	isMouseToggleEnabled = Preferences::Has("alt-mouse turning");
 	if(activeCommands.Has(Command::MOUSE_TURNING_TOGGLE))
+	{
 		isMouseToggleEnabled = !isMouseToggleEnabled;
+		Preferences::Set("alt-mouse turning", isMouseTurningEnabled);
+	}
 	// XOR mouse hold and mouse toggle. If mouse toggle is OFF, then mouse hold
 	// will temporarily turn ON mouse control. If mouse toggle is ON, then mouse
 	// hold will temporarily turn OFF mouse control.
 	isMouseTurningEnabled = (isMouseHoldEnabled ^ isMouseToggleEnabled);
-	Preferences::Set("alt-mouse turning", isMouseTurningEnabled);
 	if(!isMouseTurningEnabled)
 		return;
+	if(!activeCommands.Has(Command::MOUSE_TURNING_HOLD))
+		activeCommands.Set(Command::MOUSE_TURNING_HOLD);
```

Comment:
Not sure if I should always run `Set` or if I should detect and then Set like in this diff.  The primary point of having `Command::MOUSE_TURNING_HOLD` always set for mouse control is that it gets used in `AI.cpp` to set the turn of the ship.

Reply:
Just setting it unconditionally seems fine.